src/domain/datasets/domain/src/entities/dataset_env_var.rs

```diff
@@ -232,43 +232,46 @@ impl std::error::Error for AesGcmError {}
 #[derive(Debug, Clone, Merge, Serialize, Deserialize)]
 #[serde(deny_unknown_fields, rename_all = "camelCase")]
 pub struct DatasetEnvVarsConfig {
-    pub mode: Option<DatasetEnvVarsType>,
+    pub enabled: Option<bool>,
     /// Represents the encryption key for the dataset env vars. This field is
-    /// required if `DatasetEnvVarsType` is `Storage`.
+    /// required if `enabled` is `true` or `None`.
     ///
     /// The encryption key must be a 32-character alphanumeric string, which
     /// includes both uppercase and lowercase Latin letters (A-Z, a-z) and
     /// digits (0-9).
     ///
     /// # Example
     /// let config = DatasetEnvVarsConfig {
-    ///     mode: Some(DatasetEnvVarsType::Storage),
+    ///     enabled: Some(true),
     ///     encryption_key:
     /// Some(String::from("aBcDeFgHiJkLmNoPqRsTuVwXyZ012345")) };
     /// ```
     pub encryption_key: Option<String>,
 }
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub enum DatasetEnvVarsType {
-    Static,
-    Storage,
-}
-
 impl DatasetEnvVarsConfig {
     pub fn sample() -> Self {
         Self {
-            mode: Some(DatasetEnvVarsType::Storage),
+            enabled: Some(true),
             encryption_key: Some(SAMPLE_DATASET_ENV_VAR_ENCRYPTION_KEY.to_string()),
         }
     }
+
+    pub fn is_enabled(&self) -> bool {
+        if let Some(enabled) = self.enabled
+            && enabled
+            && self.encryption_key.is_some()
+        {
+            return true;
+        }
+        false
+    }
 }
 
 impl Default for DatasetEnvVarsConfig {
     fn default() -> Self {
         Self {
-            mode: Some(DatasetEnvVarsType::Static),
+            enabled: Some(false),
```

Comment:
If we default to `Some(false)` then the feature is disabled by default and we will never warn a user that it was not configured?

Reply:
yeah, changed